### ISSUE TITLE
similar_names complains about ipv4/ipv6

### ISSUE THREAD

**Issue by ivan** [Manishearth/rust-clippy#811]
```rust
src/session.rs:27:6: 27:12 warning: binding's name is too similar to existing binding, #[warn(similar_names)] on by default
src/session.rs:27 	let ipv6_s = if let IpAddr::V6(ip) = *remote_ip { Some(format!("{}", ip)) } else { None };
                  	    ^~~~~~
src/session.rs:26:6: 26:12 note: existing binding defined here
src/session.rs:26 	let ipv4_s = if let IpAddr::V4(ip) = *remote_ip { Some(format!("{}", ip)) } else { None };
                  	    ^~~~~~
src/session.rs:27:6: 27:12 help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#similar_names
```

Maybe these should also be whitelisted?

**Comment by mcarton** [comment]
cc @oli-obk

**Comment by oli-obk** [comment]
hmm... maybe we should simply ignore numbers? Afterwards my algorithm will complain only if both single and double digit numbers are used.

**Comment by mcarton** [comment]
Yup

**Comment by llogiq** [comment]
Also `parser` and `parsed`. Yes, it's similar, but unlikely to be erroneously swapped (different types).

**Comment by oli-obk** [comment]
those can go to the whitelist
